### PR TITLE
Add connection object

### PR DESCRIPTION
**Pull request by monnheim**
Add a top level SupplyConnection object, to indicate the physical connection to the district heating provider.

Move location information to this top level.

Make energyMeterSensorId optional at and below the HeatingSystem-level.

Add min/maxFeedTemperature on the TemperatureCurve object.

Points of uncertainty:
- The moving of the location object to be associated with a connection.
- Offsetting the entire API hierarchy by using SupplyConnection as a root object may be inconvenient to implement, given existing implementations. An option could be to assign a "connectionId" field to the HeatingSystem model and have both /heatingsystems/ and /connections/ as root objects.
- owner (which is optional) now lives on both Connection and HeatingSystem. It may be more logical to have it only on the connection level, which might however be confusing since it is the supplier that actually owns the connection.

### CODE REVIEW

File: schemas/SupplyConnection.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+type: object
+description: >
+  A connection denotes a physical connection to a district heating supplier.
+  Contains the metadata regarding the sensors on the supply-side of district heating heat exchanger.
+required:
+  - id
+  - name
+  - feedTemperatureSensorId
+  - returnTemperatureSensorId
+  - energyMeterSensorId
```
Comment (SimSmith) [inline schemas/SupplyConnection.yaml:10]: Is it likely that these values will be shared with HeatingSystem or RadiatorLoop?

Comment (monnheim) [reply schemas/SupplyConnection.yaml:10]: No, they can never be shared with them. It is possible that RadiatorLoop and HeatingSystem share a common sensor, but the SupplyConnection sensors are always different. SupplyConnection sensor measure things on the supply side of the heat exchanger, while RadiatorLoop and HeatingSystem measure things on the demand side.

File: api.yaml [hunk 2]
```diff
@@ -21,36 +22,81 @@ servers:
...
+        - $ref: "#/components/parameters/connectionIdParam"
+      responses:
+        200:
+          description: supply connections
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/SupplyConnection"
```
[the remark this answers is not in the document]
Comment (monnheim) [reply api.yaml]: Yes, but this is now removed entirely.

**Comment by monnheim** [comment]
Last commit makes some changes to make it easier to implement given existing implementations, according to comments from @henrikhansen428 over email. These are essentially of the form outlined in the second question of the original PR. 

Avoids having a /supplyconnections in the root at all. Since supplyconnection is the only thing a district heating supplier knows about, this should not be designed without talking to them first, hence we skip it for now.

File: api.yaml [hunk 3]
```diff
@@ -23,90 +22,66 @@ security:
...
-        - Supply connections
-      operationId: getSupplyConnection
-      parameters:
-        - $ref: "#/components/parameters/connectionIdParam"
+        - Heating Systems
+      operationId: listHeatingSystems
       responses:
         200:
-          description: supply connections
+          description: supply connection
```
Comment (adnilsson) [inline api.yaml:35]: I suppose this description should be reverted to say `heating systems` again.

File: api.yaml [hunk 4]
```diff
@@ -316,6 +339,13 @@ components:
       type: http
       scheme: bearer
   parameters:
+    connectionIdParam:
```
Comment (adnilsson) [inline api.yaml:342]: Where is this parameter used?

Comment (monnheim) [reply api.yaml]: Missed deleting it.

File: api.yaml [hunk 5]
```diff
@@ -67,6 +67,29 @@ paths:
...
+    get:
+      description: Get metadata for a heatingsystems supply connection to the district heating network
+      tags:
+        - Heating Systems
+      operationId: getSupplyConnection
+      parameters:
+        - $ref: '#/components/parameters/heatingSystemIdParam'
+      responses:
+        200:
+          description: heating system
```
Comment (SimSmith) [inline api.yaml:80]: Wrong description.

File: schemas/TemperatureCurve.yaml [hunk 6]
```diff
@@ -21,6 +23,14 @@ properties:
         feedTemperature: 75.0
       - outdoorTemperature: 0.0
         feedTemperature: 57.0
```
Comment (SimSmith) [inline schemas/TemperatureCurve.yaml:25]: Add a ` required` for `outdoorTemperature` and `feedTemperature` in `breakpoints.items` as well.
```
required:
  - outdoorTemperature
  - feedTemperature
```

**Pull request merged**